### PR TITLE
API Checkpoint

### LINKED ISSUE #235: Checkpoint - Can you Build an API

### PR DESCRIPTION
**Pull request by memcmahon**
I had a thought to make this kind of a mock interview.  The project is like a take-home challenge completed on day 1, and then we can do the followup interview on day 2?

### CODE REVIEW

**Comment by zoefarrell** [comment]
> I had a thought to make this kind of a mock interview. The project is like a take-home challenge completed on day 1, and then we can do the followup interview on day 2?

I like that idea a lot!! And maybe for this mock interview they know the questions that are coming and can prep for them. And then it's a live interview with one of the instructors where they have to talk through parts of their code and answer questions. (I see you already have this as the plan)

File: module4/projects/ApiCheckpoint.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+layout: page
+title: API Checkpoint
+---
+
+# Take Home Challenge
+
+A **Take-Home Challenge** is a common part of the interview process for applicants, especially beginner developers, applying for their first job in programming or software development. It is a practical task or project that will help the employer assess a candidate's technical skills, problem-solving abilities, and coding proficiency.
```
Comment (zoefarrell) [inline module4/projects/ApiCheckpoint.md:8]: LOVE this framing!

**Comment by memcmahon** [comment]
> > I had a thought to make this kind of a mock interview. The project is like a take-home challenge completed on day 1, and then we can do the followup interview on day 2?
> 
> I like that idea a lot!! And maybe for this mock interview they know the questions that are coming and can prep for them. And then it's a live interview with one of the instructors where they have to talk through parts of their code and answer questions. (I see you already have this as the plan)

Yes - love the idea of having them prep some questions.  Maybe we could do some prepped, and some surprises?  I also want them to have the chance to sweat a little 😆

File: module4/projects/ApiCheckpoint.md (hunk 1 above)
Comment (memcmahon) [reply module4/projects/ApiCheckpoint.md:8]: Chat GPT, comin' in with the assist 💪

**Comment by zoefarrell** [comment]
> > > I had a thought to make this kind of a mock interview. The project is like a take-home challenge completed on day 1, and then we can do the followup interview on day 2?
> > 
> > 
> > I like that idea a lot!! And maybe for this mock interview they know the questions that are coming and can prep for them. And then it's a live interview with one of the instructors where they have to talk through parts of their code and answer questions. (I see you already have this as the plan)
> 
> Yes - love the idea of having them prep some questions. Maybe we could do some prepped, and some surprises? I also want them to have the chance to sweat a little 😆

Love the idea of a combo, some prepped and some unexpected!

File: module4/projects/ApiCheckpoint.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+### Objective:
+The purpose of this take-home challenge is to assess your ability to design, implement, and test a RESTful API in C#/.NET that manages a collection of products associated with a "Merchant" resource.
+
+### Task Description:
+You are provided with an existing API for a **Merchant** resource, and your task is to build a C#/.NET API to manage a **Products** resource, where each product belongs to a specific merchant. The API should support basic CRUD operations (Create, Read, Update, Delete) for Products. Additionally, you need to implement XXXX statistics endpoints that provide valuable insights into the product collection as a whole, and for each merchant.
+
+### Setup
+
+* Fork and clone this [starter repo](https://github.com/turingschool-examples/Launch_Mod4TakeHomeChallenge)
```
Comment (zoefarrell) [inline module4/projects/ApiCheckpoint.md:28]: Left comments in slack, but I'll add them here too:
* When I start up the app it takes me to https://localhost:7136/weatherforecast in the browser, not sure where this is configured
* I was able to POST and create a merchant and GET and get all merchants without changing the code :tada:
* I did have to click disable SSL Verification in Postman though

File: module4/projects/ApiCheckpoint.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+You instructors will post a deadline for submissions in slack.
+
+Submit your challenge by:
+* Creating a PR from your main branch to the turingschool-examples main branch.
+* Include your Name in your PR title.
+* Include an overview of what you were able accomplish in the PR description.
+
+### Interview Time!
+
+The final step of a take-home challenge is an interview!  You will have an interview scheduled with an instructor and during that interview you will be asked questions related to APIs, Coding, and your work on the Takehome Challenge.  You *will* be asked the following questions, and you *may* be asked additional questions.
```
Comment (zoefarrell) [inline module4/projects/ApiCheckpoint.md:103]: I might clarify this language to "live interview" or something like that. I want students thinking that the code part is also part of the interview.

File: module4/projects/ApiCheckpoint.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+5. Explain the role of HTTP request bodies in REST APIs, and mention any specific C#/.NET libraries or methods you used to work with request bodies in your project.
+
+<section class='instructor-note' markdown='1'>
+
+<!-- Day 2 - Interview -->
+
+We would not all these questions with students, but my thought is that as part of this exercise, we can do a mock interview. Maybe we can make the rubric a bit more flexible - like would they have made it to the next round of the interview.  I think we can internally set a metric of completeness and their ability to answer or gracefully exit the question portion.
+
+
+1. What is a REST API, and how does it differ from other types of APIs?
```
Comment (zoefarrell) [inline module4/projects/ApiCheckpoint.md:116]: If you have any ideas for a question or two that we could ask that is something they should be able to answer that would be helpful! Or we could maybe only give two questions they will have time to prepare for?

**Review by zoefarrell — APPROVED** [review]

**Comment by zoefarrell** [comment]
I think this is going to be a great assignment to both review and push students!

**Comment by zoefarrell** [comment]
One more random thought, because we're thinking of this as a take-home assessment, I wonder how we want to frame what's ok and what's not for getting help from peers.

**Pull request merged**